File: src/main/java/com/otica_system/service/customer/CustomerService.java
package com.otica_system.service.customer;

import com.otica_system.dto.customer.CreateCustomerDTO;
import com.otica_system.dto.customer.CustomerDTO;
import com.otica_system.dto.customer.UpdateCustomerDTO;

import java.util.List;
import java.util.Optional;

public interface CustomerService {

    CustomerDTO create(CreateCustomerDTO dto);

    CustomerDTO update(Long id, UpdateCustomerDTO dto);

    void delete(Long id);

    CustomerDTO findById(Long id);

    Optional<CustomerDTO> findByCpfOptional(String cpf);

    CustomerDTO findByCpf(String cpf);

    List<CustomerDTO> findAll();
}

File: frontend/src/app/services/customer.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Customer } from '../models/customer.model';

@Injectable({
  providedIn: 'root'
})
export class CustomerService {

  private apiUrl = 'https://otica-system-backend.onrender.com/api/customers';

  constructor(private http: HttpClient) { }

  create(customer: Customer): Observable<Customer> {
    return this.http.post<Customer>(this.apiUrl, customer);
  }

  update(id: number, customer: Customer): Observable<Customer> {
    return this.http.put<Customer>(`${this.apiUrl}/${id}`, customer);
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  findById(id: number): Observable<Customer> {
    return this.http.get<Customer>(`${this.apiUrl}/${id}`);
  }

  findAll(): Observable<Customer[]> {
    return this.http.get<Customer[]>(this.apiUrl);
  }

  findByCpf(cpf: string): Observable<Customer> {
    return this.http.get<Customer>(`${this.apiUrl}/cpf?cpf=${encodeURIComponent(cpf)}`);
  }
}
